packages/core/src/types/ad.ts

```diff
@@ -39,9 +39,22 @@ import { LbfgsParams } from "./state";
 
 //#region Types for implicit autodiff graph
 
-export type VarAD = Const | Input | Unary | Binary | Ternary | Nary | Debug;
+export type Expr = Bool | VarAD | Vec;
```

Comment:
Would it make sense to drop the `AD` in `VarAD` or use `BoolAD`, `VecAD` etc.  for consistency?

Reply:
See @wodeni's earlier comment: https://github.com/penrose/penrose/pull/906#discussion_r869626946 The plan is to merge this PR first and then do a refactor from `VarAD` to `ad.Num` later.

Reply:
sorry, didn't catch that